### PR TITLE
Wip ui

### PR DESCRIPTION
**Pull request by swiftgist**
iSCSI api with runners and modules

Signed-off-by: Eric Jackson <ejackson@suse.com>

### CODE REVIEW

**Comment by jan--f** [comment]
@swiftgist can you give a hint as to how to use/test this?

File: srv/modules/runners/ui-iscsi.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+	Comments are removed from cluster.yml
+	"""
+	if 'filename' in kwargs:
+	    filename = kwargs['filename']
+	else:
+	    filename = '/srv/pillar/ceph/stack/{}/cluster.yml'.format(cluster)
+	contents = {}
+	with open(filename, 'r') as yml:
+	    contents = yaml.safe_load(yml)
```
Comment (rjfd) [inline srv/modules/runners/ui-iscsi.py]: If the file `filename` is empty or just contains comment lines then variable `contents` will have the value `None` and will throw an exception in the next line.
Please add the following to the next line:
```
if not contains:
    contents = {}

File: srv/modules/runners/ui-iscsi.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	contents['igw_config'] = 'default-ui'
+	with open(filename, 'w') as yml:
+	    yml.write(yaml.dump(contents,
+				Dumper=self.friendly_dumper,
+				default_flow_style=False))
+	# refresh pillar
+	self.local.cmd("I@roles:master", 'saltutils.pillar_refresh', [''], expr_form="compound")
+
+    def canned_populate(self, canned):
```
Comment (rjfd) [inline srv/modules/runners/ui-iscsi.py]: Shouldn't we remove these "canned_*" functions?

**Comment by swiftgist** [comment]
I added the check, unit tests and a rebase

**Comment by rjfd** [comment]
@swiftgist I approved the changes, but please merge the PR #230 (which adds some documentation to this code) before merging this PR

**Comment by swiftgist** [comment]
I left comments for #230.  I believe this PR is holding up others.

**Pull request merged**